src/python_version.rs

```diff
@@ -0,0 +1,121 @@
+use crate::runtime_txt::{self, RuntimeTxtError};
+use indoc::formatdoc;
+use libcnb::data::buildpack::StackId;
+use libherokubuildpack::log::log_info;
+use std::fmt::{self, Display};
+use std::path::Path;
+
+/// The Python version that will be installed if the project does not specify an explicit version.
+pub(crate) const DEFAULT_PYTHON_VERSION: PythonVersion = PythonVersion {
+    major: 3,
+    minor: 11,
+    patch: 2,
+};
+
+/// Representation of a specific Python `X.Y.Z` version.
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) struct PythonVersion {
+    pub major: u16,
+    pub minor: u16,
+    pub patch: u16,
+}
+
+impl PythonVersion {
+    pub fn new(major: u16, minor: u16, patch: u16) -> Self {
+        Self {
+            major,
+            minor,
+            patch,
+        }
+    }
+
+    pub fn url(&self, stack_id: &StackId) -> String {
```

Comment:
Nit, non-blocking: Personally, I would move the download URL templating out of `PythonVersion` to increase separation of concerns.

Reply:
Yeah I wasn't 100% happy with it being here (previously the URL was inlined in the layer itself) - although this is all going to change once the versions manifest file is added as part of [Automatic Python patch version updates](https://gus.lightning.force.com/lightning/r/ADM_Epic__c/a3QEE000000V4tl2AC/related/Work__r/view).